rclpy/src/rclpy/_rclpy_logging.c

```diff
@@ -0,0 +1,129 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Python.h>
+
+#include <rcutils/logging.h>
+#include <rcutils/logging_macros.h>
+#include <rcutils/time.h>
+
+/// Initialize the logging system.
+/**
+ * \return None
+ */
+static PyObject *
+rclpy_logging_initialize()
+{
+  rcutils_logging_initialize();
+  Py_RETURN_NONE;
+}
+
+/// Get the global severity threshold of the logging system.
+/**
+ * \return severity
+ */
+static PyObject *
+rclpy_logging_get_severity_threshold()
+{
+  int severity = rcutils_logging_get_severity_threshold();
+
+  return PyLong_FromLong(severity);
+}
+
+/// Set the global severity threshold of the logging system.
+/**
+ *
+ * \param[in] severity Threshold to set
+ * \return None, or
+ * \return NULL on failure
```

Comment:
updated in 71ab9a92bb304390bda4a48cf11b20ce9750d58f

Reply:
Sorry that was unclear in my previous comment, I meant that if we want to document the behavior on failure we should document the error that is raised rather than the return value (so that users can catch it if they want)

Reply:
I think I'm missing something. It's not raising an error though, so "None or NULL on failure" seems to be the appropriate documentation for the return value from what I can tell? perhaps what you're suggesting is that it should raise?

Reply:
PyArg_ParseTuple will raise for you (https://docs.python.org/3.5/c-api/arg.html#c.PyArg_ParseTuple) so the user will never see the return code

Reply:
Ah, I see now, thanks. In general we don't document that exception so I agree with your original point that it's not necessary to explain that in the docblock: 226c96e0c7d664aedbbe76a92ba0747fdc393a28